File: c++/structures/structures.cpp
#include <iostream>

struct MoneyBox {
  std::string name = "";
  double saved_amount = 0;

  void save(double amount) {
    saved_amount += amount;
  }
};

int main() {
  MoneyBox piggy;
  piggy.name = "Miss Piggy";

  std::cout << "Money box name: " << piggy.name << std::endl;
  std::cout << "Money box saved amount: " << piggy.saved_amount << std::endl;

  piggy.save(20);

  std::cout << "Money box saved amount: " << piggy.saved_amount << std::endl;

  piggy.save(100);

  std::cout << "Money box saved amount: " << piggy.saved_amount << std::endl;
}

File: c#/structures/structures.cs
struct MoneyBox {
  public string name;
  public double saved_amount;

  public void save(double amount) {
    saved_amount += amount;
  }
};

class Structures {
  public static void Main(string[] args) {
    var piggy = new MoneyBox();
    piggy.name = "Miss Piggy";

    System.Console.WriteLine("Money box name: " + piggy.name);
    System.Console.WriteLine("Money box saved amount: " + piggy.saved_amount);

    piggy.save(20);

    System.Console.WriteLine("Money box saved amount: " + piggy.saved_amount);

    piggy.save(100);

    System.Console.WriteLine("Money box saved amount: " + piggy.saved_amount);
  }
}
